test/requirements.in

```diff
@@ -16,6 +16,7 @@ ruff>=0.0.260
 respx
 scikit-image
 shap
+spaces; python_version >= '3.9'
```

Comment:
Why do we add `spaces` to the test requirements when it is not actually used in any tests?

Reply:
It is needed so typechecking (pyright) doesn't error out

I can still `# type: ignore` the problematic line instead if you are trying to keep the test requirements as small as possible

Reply:
Yup that'd be better thanks!